Update References to the new Technische Richtlinien (#282)
sehlen-bsi/botan-docs: docs/cryptodoc/src/05_07_slh_dsa.rst

(SLH-DSA) is found in
:srcref:`src/lib/pubkey/sphincsplus/` and follows [FIPS-205]_.

BSI's [TR-02102-1]_ recommends using SLH-DSA in "hedged" mode with the parameter

> sets that fulfill NIST's category 3 and 5 ("192" or "256" both fast and small).

Reviewer: both fast and small and both types of hashes?

Author: Yes. Adapted to make this more clear.

Hash-based signatures are considered secure for long-term secure authentication

> and do not need to be used in a hybrid form with a classical signature method.

Reviewer: ```suggestion
and are not required to be used in a hybrid form with a classical signature method.
```
nit, just to sound a bit more like the TR.
